### PR TITLE
Fix: Workflows, eslint rules, build

### LINKED ISSUE #36: [BUG] Lint workflow crash
### Description of the bug

The workflow file [lint.yml](https://github.com/WebXDAO/DevConf/blob/main/.github/workflows/lint.yml) is bugged.

The 3 actions throw `command not found`:
- ESLint: `error Command "lint:strict" not found.`
- Typecheck: `error Command "typecheck" not found.`
- Prettier: `error Command "format:check" not found.`

See more errors examples [here](https://github.com/WebXDAO/DevConf/actions/runs/3630283680/jobs/6123453450).

### Please add screenshots (if applicable)

_No response_

### Add any other context about the problem here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Chandraprakash-Darji**
## Related Issue

1. eslint: plugin updates and rules
2. pretteier: add .next to prettierignore 
3. nextcongif: specified the eslint dir to src
4. deps: added the format and lint scripts
5. tsconfig: add the path alias
6. .vscode: the list of extensions and settings

Closes: #36 

### Describe the changes you've made

- add the required package and script to run workflows
- pretteier will ignore the .next folder when eslint and format runs
- add the 'src' to eslint in .next.config
- Add the vscode settings that what action will be perfom on save

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update.
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly wherever it was hard to understand.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots (if applicable)

|      Original       |      Updated       |
| :-----------------: | :----------------: |
| original screenshot | updated screenshot |

## Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/WebXDAO/.github/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +36,8 @@ yarn-error.log*
 # typescript
 *.tsbuildinfo
 next-env.d.ts
+
+# next-sitemap
+robots.txt
+sitemap.xml
+sitemap-*.xml
```
Comment (Panquesito7) [inline .gitignore]: Missing an endline.
```suggestion
sitemap-*.xml

```

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (Panquesito7) [inline .vscode/extensions.json:1]: We can add these to the Gitpod file. Would that be good?

Comment (Chandraprakash-Darji) [reply .vscode/extensions.json:1]: It would be great.

File: .gitignore (hunk 1 above)
Comment (Chandraprakash-Darji) [reply .gitignore]: Is it necessary?

**Comment by Chandraprakash-Darji** [comment]
As I push all file changes according to the prettier and eslint all the workflow will run well.

**Comment by Chandraprakash-Darji** [comment]
Actually it is running all the workflows.

**Review by vinzvinci — APPROVED** [review]
LGTM! I think Gitpod can be added in the next PR @Panquesito7 .

**Review by mkubdev — APPROVED** [review]
LGTM!

**Pull request merged**